src/algebraic_geometry/prime_spectrum/noetherian.lean

```diff
@@ -90,4 +91,25 @@ begin
   { rintro (hx | hy); contradiction },
 end
 
+open topological_space
+
+instance : noetherian_space (prime_spectrum R) :=
+begin
+  rw (noetherian_space_tfae $ prime_spectrum R).out 0 1,
+  have H : is_noetherian_ring R := infer_instance,
+  rw [is_noetherian_ring_iff, is_noetherian_iff_well_founded] at H,
+  have : (closeds (prime_spectrum R))ᵒᵈ ↪o ideal R :=
+  { to_fun := λ s, vanishing_ideal ↑(show closeds $ prime_spectrum R, from s),
+    inj' := λ s t e, begin
+      apply_fun zero_locus ∘ (coe : _ → set R) at e,
+      dsimp at e,
+      ext1,
+      simpa only [zero_locus_vanishing_ideal_eq_closure, closure_eq_iff_is_closed.mpr s.closed,
+        closure_eq_iff_is_closed.mpr t.closed] using e,
+    end,
+    map_rel_iff' := λ s t, by { dsimp, rw [← subset_zero_locus_iff_le_vanishing_ideal,
+      zero_locus_vanishing_ideal_eq_closure, closure_eq_iff_is_closed.mpr s.closed], refl } },
```

Comment:
I still suggest that you add the lemma [I suggested](https://github.com/leanprover-community/mathlib/pull/14965/files/bf0f963bdc2bb83b8026022770dd818d037e00a3#r907983029):
```
import algebraic_geometry.prime_spectrum.basic
open prime_spectrum
theorem vanishing_ideal_anti_mono_iff {R : Type*} [comm_ring R]
  {s t : set (prime_spectrum R)} (hs : is_closed s) (ht : is_closed t) :
  s ⊆ t ↔ prime_spectrum.vanishing_ideal t ≤ prime_spectrum.vanishing_ideal s :=
⟨vanishing_ideal_anti_mono, λ h,
begin
  rw [← hs.closure_eq, ← ht.closure_eq],
  convert ← zero_locus_anti_mono_ideal h;
  apply zero_locus_vanishing_ideal_eq_closure,
end⟩
```
and then you can use [order_embedding.of_map_le_iff](https://leanprover-community.github.io/mathlib_docs/order/hom/basic.html#order_embedding.of_map_le_iff). It may be worth extracting the order embedding as a def as well.

There is a function [order_dual.of_dual](https://leanprover-community.github.io/mathlib_docs/order/synonym.html#order_dual.of_dual) to convert between the order_dual and the original type, which doesn't abuse defeq and break API barrier; so I would write this definition as follows:
```
  { to_fun := λ s, vanishing_ideal s.of_dual,
    inj' := λ s t e, begin
      apply_fun zero_locus ∘ (coe : _ → set R) at e,
      rw ← order_dual.of_dual_inj, ext1,
      simpa only [function.comp_app, zero_locus_vanishing_ideal_eq_closure,
        closure_eq_iff_is_closed.mpr s.of_dual.closed,
        closure_eq_iff_is_closed.mpr t.of_dual.closed] using e,
```
(but with the lemma above you don't need this).